oauth2/src/main/java/com/linecorp/armeria/common/auth/oauth2/TokenRequestException.java

```diff
@@ -0,0 +1,87 @@
+
+  /**
+   * Constructs a new {@link TokenRequestException} using {@code errorDescription} and {@code errorUri}.
+   * @param errorDescription OPTIONAL. Human-readable ASCII [USASCII] text providing additional information,
+   *                         used to assist the client developer in understanding the error that occurred.
+   *                         Values for the {@code errorDescription} parameter MUST NOT include
+   *                         characters outside the set {@code %x20-21} / {@code %x23-5B} / {@code %x5D-7E}.
+   * @param errorUri OPTIONAL. A URI identifying a human-readable web page with information about the error,
+   *                 used to provide the client developer with additional information about the error.
+   *                 Values for the {@code errorUri} parameter MUST conform to the URI-reference syntax and
+   *                 thus MUST NOT include characters outside
+   *                 the set {@code %x21} / {@code %x23-5B} / {@code %x5D-7E}.
+   */
+  public TokenRequestException(String errorDescription, @Nullable String errorUri) {
+    super(errorDescription);
+    this.errorUri = errorUri;
+  }
+
+  /**
+   * Constructs a new {@link TokenRequestException} using {@code errorDescription} and {@code errorUri}.
+   * @param errorDescription OPTIONAL. Human-readable ASCII [USASCII] text providing additional information,
+   *                         used to assist the client developer in understanding the error that occurred.
+   *                         Values for the {@code errorDescription} parameter MUST NOT include
+   *                         characters outside the set {@code %x20-21} / {@code %x23-5B} / {@code %x5D-7E}.
+   * @param errorUri OPTIONAL. A URI identifying a human-readable web page with information about the error,
+   *                 used to provide the client developer with additional information about the error.
+   *                 Values for the {@code errorUri} parameter MUST conform to the URI-reference syntax and
+   *                 thus MUST NOT include characters outside
+   *                 the set {@code %x21} / {@code %x23-5B} / {@code %x5D-7E}.
+   * @param cause the cause (which is saved for later retrieval by the {@link #getCause()} method).
+   *              (A {@code null} value is permitted, and indicates that the cause is nonexistent or unknown.)
+   */
+  public TokenRequestException(String errorDescription, @Nullable String errorUri, Throwable cause) {
+    super(errorDescription, cause);
+    this.errorUri = errorUri;
+  }
+
+  /**
+   * A URI identifying a human-readable web page with information about the error, used to provide the client
+   * developer with additional information about the error.
+   * Values for the {@code errorUri} parameter MUST conform to the URI-reference syntax and thus MUST NOT
+   * include characters outside the set {@code %x21} / {@code %x23-5B} / {@code %x5D-7E}.
+   */
+  @Nullable
+  public String getErrorUri() {
```

Comment:
How about accepting `URI` when constructing this exception and returning a `URI`?

Reply:
This URI comes in the error response body (as per the spec). I did not want to parse it. What if it's invalid or corrupt? Then we will fail processing the exception with another exception.